install-yarn.js

```diff
@@ -6,13 +6,9 @@ var yarnBin
   , node = process.argv[0]
   ;
 
-if (process.env['npm_execpath'] && process.env['npm_execpath'].match(/node_modules[\/\\]yarn[\/\\]bin[\/\\]yarn\.js$/)) {
+if (process.env['npm_execpath'] && process.env['npm_execpath'].match(/[\/\\]yarn(-v\d+\.\d+\.\d+)?[\/\\]bin[\/\\]yarn\.js$/)) {
   var execPath = process.env['npm_execpath'];
-  var expectedPath = path.join('yarn', 'bin', 'yarn.js');
```

Comment:
Can't you modify this to `path.join('bin', 'yarn.js');` and leave the rest of the code the way it is?

Reply:
if you want, but it's the same thing as the regex

Reply:
The thing is I'm not sure what the goal of

```
if (execPath.slice(-1 * expectedPath.length) === expectedPath) {
```

is and I don't want to introduce a regression by removing it.